packages/xod-arduino/platform/dtoa.h

```diff
@@ -0,0 +1,150 @@
+/**
+ * Provide `dtoa` cross-platform number to string converter function.
+ * Taken from here: https://github.com/client9/stringencoders/blob/master/src/modp_numtoa.c
+ * Original function name: `modp_dtoa2`.
+ *
+ * Modified:
+ * - `isnan` instead of tricky comparing and return "NaN"
+ * - handle Infinity values and return "Inf" or "-Inf"
+ * - return `OVF` and `-OVF` for numbers bigger than max possible, instead of using `sprintf`
+ * - use `xod::Number` instead of double
+ * - if negative number rounds to zero, return just "0" instead of "-0"
+ *
+ * This is a replacement of `dtostrf`.
+ */
+
+/**
+ * Powers of 10
+ * 10^0 to 10^9
+ */
+static const xod::Number powers_of_10[] = { 1, 10, 100, 1000, 10000, 100000, 1000000,
+    10000000, 100000000, 1000000000 };
+
+static void strreverse(char* begin, char* end)
+{
+    char aux;
+    while (end > begin)
+        aux = *end, *end-- = *begin, *begin++ = aux;
+};
+
+size_t dtoa(xod::Number value, int prec, char* str)
+{
```

Comment:
I think `dtoa` is not the best name. `d` stands for `double`, but `Number` is not equivalent to `double`. So, it really should be… why overcomplicate… `formatNumber`.

Also, I suggest enclosing the contents with `namespace xod { }` and header file guards. Prefixing `Number` with `xod::` will no longer be required. See `listFuncs.h` for the reference.

Also, would you pass the file through `clang-format`?